### PR TITLE
RD-3222 Remove globally disabled ESLint rules

### PR DESCRIPTION
**Pull request by leon-good-life**
## Description

Updated `cloudify-ui-common` version to use new common ESLint configuration. Updated local ESLint configuration and removed rule disable comments from the code accordingly.

Fixed some audit issues using `npm audit fix` at once.

## Screenshots / Videos
N/A

## Checklist
- [x] My code follows the UI Code Style.
- [x] I verified that all tests and checks have passed.
- [x] I verified if that change requires a documentation update.
- [x] I added proper labels to this PR.

## Tests
See `npm run lint` in CI pipeline.
[Stage-UI-System-Test/2088](https://jenkins.cloudify.co/view/UI%20Health%20View/job/Stage-UI-System-Test/2088/) - only Filters spec failed (known issue - RD-4286), so I consider it as PASSED

## Documentation
N/A

### CODE REVIEW

**Review by NorbertAmartus — APPROVED** [review]
Nice one! 🎉

**Review by kubama — COMMENTED** [review]
Looks fine to me.

I assume changes you made to eslint config files won't be necessary once changes are done in `cloudify-ui-common`.

**Comment by leon-good-life** [comment]
**TODO:**
- Merge the PR in `cloudify-ui-common`, then release a version and install it in this repository and in `composer`.

**Review by qooban — COMMENTED** [review]
I'm opting out from the review (at least in this week).

**Review by jawamartus — COMMENTED** [review]
LGFM

**Review by NorbertAmartus — APPROVED** [review]

**Review by jawamartus — APPROVED** [review]
LGTM

**Comment by leon-good-life** [comment]
@kubama 

> Looks fine to me.
> I assume changes you made to eslint config files won't be necessary once changes are done in cloudify-ui-common.

The rules still necessary. I tried to remove it:
Screen Shot 2022-02-09 at 16 49 47

And it caused warnings and errors in the terminal when running `rpm run devServer`:
Screen Shot 2022-02-09 at 16 49 19

**Comment by leon-good-life** [comment]
@kubama 
> This PR still needs switching to the newly released cloudify-ui-common version.

I made another PR in which cloudily-ui-common updated: https://github.com/cloudify-cosmo/cloudify-stage/pull/1806

and I pulled master to this branch

**Comment by kubama** [comment]
@leon-good-life This PR updates three `.eslintrc.json` files in which the rules are disabled explicitly. What's the point if the rules are already disabled globally in cloudify-ui-common?

File: app/.eslintrc.json [hunk 1]
```diff
@@ -6,7 +6,10 @@
     ],
```
Comment (leon-good-life) [inline app/.eslintrc.json:6]: in this configuration file only `"../node_modules/cloudify-ui-common/configs/eslint-common-react.json"` is extended so the rule `"no-void": "off",` no applied and it is necessary to add it here. Or to add extend `"./node_modules/cloudify-ui-common/configs/eslint-common.json"`.

Comment (kubama) [reply app/.eslintrc.json:6]: Please see `eslint-common-react.json` - it already extends `eslint-common.json`, so it really should work.

Comment (leon-good-life) [reply app/.eslintrc.json:6]: Some files are extended in both places in `cloudify-ui-common`, like `"./eslint-ts-overrides.json"` here:
Screen Shot 2022-02-14 at 14 04 27

Comment (jawamartus) [reply app/.eslintrc.json:6]: It looks like the extends of `./eslint-ts-overrides.json` is redundant inside of `eslint-common-react.json`.

**Comment by qooban** [comment]
As agreed with @leon-good-life I'll take over this PR.

**Review by qooban — COMMENTED** [review]
Ready for re-review.

File: .eslintrc.json [hunk 2]
```diff
@@ -30,7 +14,8 @@
                     }
                 ],
                 "no-console": "off",
-                "security/detect-child-process": "off"
+                "security/detect-child-process": "off",
+                "no-process-exit": "off"
```
Comment (kubama) [inline .eslintrc.json:18]: I'm not sure why this rule disable was added, I can see no other reference to this rule in the diff.

File: app/.eslintrc.json [hunk 3]
```diff
@@ -6,7 +6,8 @@
...
     "rules": {
         "no-console": "error",
-        "prefer-promise-reject-errors": "off"
+        "prefer-promise-reject-errors": "off",
+        "@typescript-eslint/no-namespace": "off"
```
Comment (kubama) [inline app/.eslintrc.json:10]: This rule is being disabled here, but before the change it was enabled in the top-level `.eslintrc.json`. However, I can see no other changes related to this rule (like removing explicit disables in the source files) - is this OK?

File: .eslintrc.json (hunk 2 above)
Comment (qooban) [reply .eslintrc.json:26]: Right now this `overrides` section extends "./node_modules/cloudify-ui-common/configs/eslint-common-node.json", which by default enables that rules. This `overrides` section affects only files in the root directory and `scripts` folder, so only our internal scripts. We use `process.exit` there and I think it is fine, so I decided to add `"no-process-exit": "off"`.

File: app/.eslintrc.json (hunk 3 above)
Comment (qooban) [reply app/.eslintrc.json]: Good catch! 🙂 My mistake. I reverted the rule to be in the original place and in the original shape. No need to change it.

**Review by kubama — APPROVED** [review]

**Pull request merged**